src/Umbraco.Web.BackOffice/Controllers/MediaController.cs

```diff
@@ -576,252 +576,258 @@ public async Task<IActionResult> PostSort(ContentSortOrder sorted)
 
-            IMedia createdMediaItem = _mediaService.CreateMedia(mediaItemName, parentId.Value, mediaTypeAlias,
-                _backofficeSecurityAccessor.BackOfficeSecurity?.CurrentUser?.Id ?? -1);
+                await using (Stream stream = formFile.OpenReadStream())
+                {
+                    createdMediaItem.SetValue(_mediaFileManager, _mediaUrlGenerators, _shortStringHelper,
+                        _contentTypeBaseServiceProvider, Constants.Conventions.Media.File, fileName, stream);
+                }
 
-            await using (Stream stream = formFile.OpenReadStream())
-            {
-                createdMediaItem.SetValue(_mediaFileManager, _mediaUrlGenerators, _shortStringHelper,
-                    _contentTypeBaseServiceProvider, Constants.Conventions.Media.File, fileName, stream);
+                Attempt<OperationResult?> saveResult = _mediaService.Save(createdMediaItem,
+                    _backofficeSecurityAccessor.BackOfficeSecurity?.CurrentUser?.Id ?? -1);
+                if (saveResult == false)
+                {
+                    AddCancelMessage(tempFiles,
+                        _localizedTextService.Localize("speechBubbles", "operationCancelledText") + " -- " + mediaItemName);
+                }
             }
 
-            Attempt<OperationResult?> saveResult = _mediaService.Save(createdMediaItem,
-                _backofficeSecurityAccessor.BackOfficeSecurity?.CurrentUser?.Id ?? -1);
-            if (saveResult == false)
+            //Different response if this is a 'blueimp' request
+            if (HttpContext.Request.Query.Any(x => x.Key == "origin"))
             {
-                AddCancelMessage(tempFiles,
-                    _localizedTextService.Localize("speechBubbles", "operationCancelledText") + " -- " + mediaItemName);
+                KeyValuePair<string, StringValues> origin = HttpContext.Request.Query.First(x => x.Key == "origin");
+                if (origin.Value == "blueimp")
+                {
+                    _postAddFileSemaphore.Release();
+                    return new JsonResult(tempFiles); //Don't output the angular xsrf stuff, blue imp doesn't like that
+                }
             }
-        }
 
-        //Different response if this is a 'blueimp' request
-        if (HttpContext.Request.Query.Any(x => x.Key == "origin"))
-        {
-            KeyValuePair<string, StringValues> origin = HttpContext.Request.Query.First(x => x.Key == "origin");
-            if (origin.Value == "blueimp")
-            {
-                _postAddFileSemaphore.Release();
-                return new JsonResult(tempFiles); //Don't output the angular xsrf stuff, blue imp doesn't like that
-            }
+            _postAddFileSemaphore.Release();
+            return Ok(tempFiles);
+        }catch(Exception ex){
+            _logger.LogError( ex,"Something went wrong adding files");
+            _postAddFileSemaphore.Release();
+            throw;
         }
```

Comment:
```suggestion
        catch (Exception ex)
        {
            _logger.LogError(ex, "Something went wrong adding files");
            throw;
        }
        finally
        {
            _postAddFileSemaphore.Release();
        }
```
I think it would be great to put the `_postAddFileSemaphore.Release();` in a `finally`  so that it's always executed no matter what.